src/components/course/ManageCoursePage.js

```diff
@@ -0,0 +1,119 @@
+import React, {PropTypes} from 'react';
+import {connect} from 'react-redux';
+import {bindActionCreators} from 'redux';
+import * as courseActions from '../../actions/courseActions';
+import CourseForm from './CourseForm';
+import toastr from 'toastr';
+import {browserHistory} from 'react-router';
+
+export class ManageCoursePage extends React.Component {
+  constructor(props, context) {
+    super(props, context);
+
+    this.state = {
+      course: Object.assign({}, this.props.course),
+      errors: {},
+      saving: false
+    };
+
+    this.saveCourse = this.saveCourse.bind(this);
+    this.deleteCourse= this.deleteCourse.bind(this);
+    this.updateCourseState = this.updateCourseState.bind(this);
+  }
+  componentWillReceiveProps(nextProps) {
+    if (this.props.course.id != nextProps.course.id) {
+      this.setState({course: Object.assign({}, nextProps.course)});
+    }
+  }
+
+  updateCourseState(event) {
+    const field = event.target.name;
+    let course = Object.assign({}, this.state.course);
+    course[field] = event.target.value;
+    return this.setState({course: course});
+  }
+
+  deleteCourse(event){
+    event.preventDefault();
+    this.props.actions.deleteCourse(this.state.course.id);
+    toastr.error(this.state.course.id + ' is Deleted');
+    browserHistory.push('/courses');
+    
+  }
+  
+  saveCourse(event) {
+    event.preventDefault();
+    this.setState({saving:true});
+    this.props.actions.saveCourse(this.state.course)
+        .then(()=>this.redirect());
+       
+}
+
+redirect(){
+    this.setState({saving:false});
+    toastr.success('Course Added');
+    this.context.router.push('/courses'); 
+}
+
+  render() {
+    return (
+
+      <CourseForm
+        course={this.state.course}
+        onChange={this.updateCourseState}
+        onSave={this.saveCourse}
+        errors={this.state.errors}
+        allAuthors={this.props.authors}
+        onDelete={this.deleteCourse}
+        saving={this.state.saving}
+      />
+    );
+  }
+}
+
+ManageCoursePage.propTypes = {
+  course: PropTypes.object.isRequired,
+  authors: PropTypes.array.isRequired,
+  actions: PropTypes.object.isRequired
+
+};
+
+ManageCoursePage.contextTypes = {
+  router: PropTypes.object
+};
+
+const getCourseById = (courses, id) => {
+    const course = courses.filter(course => course.id == id);
```

Comment:
Since we're looking for a single course. find seems like a better method to use. the filter would go through all the values in the array to find a match, also you should mind the `==`. make sure to use `=== ` as that would do value match and not a type match. eg 
```"01" == 01 // true  `"01" === 01 // false ```

I would suggest you use find ```courses.find(course => course.id === id);``` this would return either the course or ```null```

```
const getCoursebyId = (courses, id) => courses.find(course => course.id === id)
```
This looks cleaner simpler and will terminate as soon as it finds the course. That makes it optimal